tensorflow/python/ops/gradients.py

```diff
@@ -829,3 +829,45 @@ def _hessian_vector_product(ys, xs, v):
 
   # Second backprop
   return gradients(elemwise_products, xs)
+
+
+def hessian(ys, x, name="hessian", colocate_gradients_with_ops=False, 
+            gate_gradients=False, aggregation_method=None):
+  """Constructs the Hessian of sum of `ys` with respect to `x`.
+
+  `hessian()` adds ops to the graph to output the Hessian of `ys` with 
+  respect to `x` and returns a `Tensor`.
+
+  Args:
+    ys: A `Tensor` or list of tensors to be differentiated.
+    x: A `Tensor` to be used for differentiation.
+    name: Optional name to use for grouping all the gradient ops together.
+      defaults to 'gradients'.
+    colocate_gradients_with_ops: If True, try colocating gradients with
+      the corresponding op.
+    gate_gradients: If True, add a tuple around the gradients returned
+      for an operation.  This avoids some race conditions.
+    aggregation_method: Specifies the method used to combine gradient terms.
+      Accepted values are constants defined in the class `AggregationMethod`.
+
+  Returns:
+    The hessian of `sum(ys)` with respect to `x`.
+
+  Raises:
+    LookupError: if one of the operations between `x` and `ys` does not
+      have a registered gradient function.
+    ValueError: if the arguments are invalid
+  """
+  ndims = x.get_shape().ndims
+  if ndims is not None and ndims != 1:
+    raise ValueError("Hessians can only be computed with respect to one-dimensional tensors")
```

Comment:
Thanks for the comments @goodfeli, and thanks for the follow ups @tillahoffmann !  So is this something that we need a new function / interface for?  Is it too hard for users to just call gradients twice instead?

In general, TensorFlow tries to promote composition when possible, and only provides higher level interfaces when it is an extremely common pattern, or when there's something particularly tricky to get correct and we don't want users shooting themselves in the foot.  From what I can tell, this function doesn't really fit either of those criteria, but I could be convinced otherwise.

My inclination is to avoid adding this function just yet, mostly to keep maintenance burden low, unless we get an large number of people who think this is useful to have.  Thoughts?